packetbeat/flows/worker.go

```diff
@@ -213,8 +264,12 @@ func makeWorker(processor *flowsProcessor, tick time.Duration, timeout, period i
 			if nPeriod <= 0 {
 				nPeriod = period
 			}
+			handleActiveTimeout := enableActiveFlowTimeout && nActiveTimeout == 0
```

Comment:
what is `nActiveTimeout` and why might it be 0?

Reply:
tick will be gcd of period, timeout and activetimeout
so if period = 10, timeout = 30 and activetimeout = 60
tick = 10
Initial nPeriod = 1
Initial ntimeout = 3
Initial nActiveTimeout = 6

At each tick we decrement these values and then when any of these values reach 0 we either report flow, check for timeout or check for active timeout.

Now if say activetimeout is not set, or period is set to -1

then tick = timeout
nperiod = -1 
nActivetimeout = -1

we will never decrement nperiod or nActivetimeout during the ticks, they will never reach 0 so we will never check for period or active timeout.

I added unit tests around this. Maybe that helps a little bit